pkgs/development/python-modules/gradio/default.nix

```
@@ -0,0 +1,140 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, writeTextFile
+, setuptools
+, analytics-python
+, aiohttp
+, fastapi
+, ffmpy
+, markdown-it-py#[linkify,plugins]
+, linkify-it-py
+, mdit-py-plugins
+, matplotlib
+, numpy
+, orjson
+, pandas
+, paramiko
+, pillow
+, pycryptodome
+, python-multipart
+, pydub
+, requests
+, uvicorn
+, jinja2
+, fsspec
+, httpx
+, pydantic
+, typing-extensions
+, pytest-asyncio
+, mlflow
+, huggingface-hub
+, transformers
+, wandb
+, respx
+, scikitimage
+, shap
+, ipython
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "gradio";
+  version = "3.1.4";
+  disabled = pythonOlder "3.7";
+
+  # We use the Pypi release, as it provides prebuild webui assets,
+  # and its releases are also more frequent than github tags
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "5pzOfHDEBdz+HNGoaPwimBKixAZ1K6zD1a0IWLfzfxo=";
+  };
+
+  propagatedBuildInputs = [
+    analytics-python
+    aiohttp
+    # h11 # not needed, https://github.com/gradio-app/gradio/pull/1680
+    fastapi
+    ffmpy
+    markdown-it-py
+  ] ++ markdown-it-py.optional-dependencies.plugins
+    ++ markdown-it-py.optional-dependencies.linkify
+    ++ [
+    matplotlib
+    numpy
+    orjson
+    pandas
+    paramiko
+    pillow
+    pycryptodome
+    python-multipart
+    pydub
+    requests
+    uvicorn
+    jinja2
+    fsspec
+    httpx
+    pydantic
+  ];
+
+  postPatch = ''
+    # Unpin h11, as its version was only pinned to aid dependency resolution.
+    # Basically a revert of https://github.com/gradio-app/gradio/pull/1680
+    substituteInPlace requirements.txt \
+      --replace "h11<0.13,>=0.11" ""
+  '';
+
+  checkInputs = [
+    pytestCheckHook
+    pytest-asyncio
+    mlflow
+    #cometml # FIXME: enable once comet-ml is packaged
+    huggingface-hub
+    transformers
+    wandb
+    respx
+    scikitimage
+    shap
+    ipython
+   ];
+  disabledTestPaths = [
+    # touches network
+    "test/test_tunneling.py"
+    "test/test_external.py"
+  ];
+  disabledTests = [
+    # touches network
```

```suggestion
    # requires network connection
```
because local networking would work in tests